### ISSUE TITLE
Floats validate as integers

### ISSUE THREAD

**Issue by gitlabr** [CZ-NIC/yangson#47]
Similarly to https://github.com/CZ-NIC/yangson/issues/37, yangson considers floats to be valid values for integer type nodes.

Shamelessly stealing the minimal example from the linked issue:
```
$ cat test.json 
{
    "ietf-yang-library:modules-state": {
      "module-set-id": "",
      "module": [
        {
          "name": "test",
          "revision": "1.0.0",
          "conformance-type": "implement"
        }
      ]
    }
  }
$ cat test.yang 
module test {
    prefix ;
    revision 1.0.0;
    description "Test schema.";
    leaf number {
        type int32;
    }
}
```

I would expect the following code to raise an exception, but it doesn't:

```
from yangson import DataModel
import yaml

data = DataModel.from_file("./test.json")
yaml_input = yaml.safe_load("'test:number': 3.14")
inst = data.from_raw(yaml_input)
inst.validate()
```

**Comment by gitlabr** [comment]
It looks like string values containing integers, e.g. `"3"` also validate incorrectly.

**Comment by llhotka** [comment]
Fixed with 1c60783a4db8f8153a833707b69dc8cbe64ba88e.
